lib/astarte_core/triggers/policy/tagged_policy.ex

```diff
@@ -0,0 +1,116 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2021 Ispirata Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.Core.Triggers.Policy.TaggedPolicy do
+  use Astarte.Core.Triggers.PolicyProtobuf
+  alias Astarte.Core.Triggers.Policy
+  alias Astarte.Core.Triggers.Policy.Handler
+  alias Astarte.Core.Triggers.Policy.TaggedPolicy
+
+  defstruct name: "",
+            policy_container: nil
+
+  @doc """
+  Creates a `TaggedPolicy` from a `Policy`.
+
+  It is assumed that the `Policy` is valid and constructed using `Policy.changeset`
+
+  Returns a `%TaggedPolicy{}`
+  """
+  def from_policy(%Policy{} = policy) do
+    %Policy{
+      name: name,
+      error_handlers: error_handlers,
+      maximum_capacity: maximum_capacity,
+      retry_times: retry_times,
+      event_ttl: event_ttl
+    } = policy
+
+    %TaggedPolicy{
+      name: name,
+      policy_container:
+        to_policy_container(error_handlers, maximum_capacity, retry_times, event_ttl)
+    }
+  end
+
+  defp to_policy_container(error_handlers, maximum_capacity, retry_times, event_ttl) do
+    PolicyContainer.new(
+      maximum_capacity: maximum_capacity,
+      retry_times: retry_times,
+      event_ttl: event_ttl || 0,
+      error_handlers: Enum.map(error_handlers, &Handler.to_tagged_handler/1)
+    )
+  end
+
+  @doc """
+  Deserializes a `%TaggedPolicy{}` from `db_result`.
+  `db_result` can be a keyword list or a map.
+
+  Returns the `{:ok, %TaggedPolicy{}}` on success,
+  `{:error, :invalid_tagged_policy_data}` on failure.
+  """
+  def from_db_result(db_result) when not is_map(db_result) do
+    db_result
+    |> Enum.into(%{})
+    |> from_db_result()
+  end
+
+  def from_db_result(db_result) do
+    with %{
+           name: name,
+           policy_container: policy_container
+         } <- db_result do
+      tagged_policy = %TaggedPolicy{
+        name: name,
+        policy_container: PolicyContainer.decode(policy_container)
+      }
+
+      {:ok, tagged_policy}
+    else
+      _ ->
+        {:error, :invalid_tagged_policy_data}
+    end
+  end
```

Comment:
Your DB result here will just be a binary with the serialized trigger, this should probably become something like
[this](https://github.com/astarte-platform/astarte_core/blob/master/lib/astarte_core/triggers/simple_triggers_protobuf/utils.ex#L64)